tests/transformers/test_transformers_model_export.py

```diff
@@ -3411,8 +3413,11 @@ def _calculate_expected_size(path_or_dir):
     ],
 )
 def test_text_generation_save_model_with_inference_task(
-    task, input_example, text_generation_pipeline, model_path
+    monkeypatch, task, input_example, text_generation_pipeline, model_path
 ):
+    # Strictly raise error during requirements inference for testing purposes
```

Comment:
note: Ideally we apply this to all test cases, but it actually caused failure for some test cases. Will follow-up in a separate PR.

Reply:
I imagine that an audit for version compatibility will be in order for ensuring that we can support this accurate requirements inference - let's file a JIRA for this (3pt) since we should also do some refactoring of the massive serialization suite at the same time in order to split up some of these tests to logical groupings

Reply:
SG!